pkg/kubelet/kubelet.go

```diff
@@ -867,9 +867,9 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	}
 	klet.AddPodSyncLoopHandler(activeDeadlineHandler)
 	klet.AddPodSyncHandler(activeDeadlineHandler)
-	if utilfeature.DefaultFeatureGate.Enabled(features.TopologyManager) {
-		klet.admitHandlers.AddPodAdmitHandler(klet.containerManager.GetTopologyPodAdmitHandler())
-	}
+
+	klet.admitHandlers.AddPodAdmitHandler(klet.containerManager.GetAllocateResourcesPodAdmitHandler())
```

Comment:
Is it guaranteed that the AdmitHandler is called on every occasion, for example on a Kubelet restart? I'm thinking if it's ever possible to have no allocated resources when a container is added.

Reply:
Yes. Admit handler is always called -- even across a kubelet restart.

Reply:
I agree though, we will need to make sure (especially in the CPUManager), that we handle the fact that Allocate() can be called multiple times for the same container.

Reply:
@nolancon ^^

Reply:
This seems to be already handled:  https://github.com/kubernetes/kubernetes/blob/aa964c905576d3e873994de156999f609ae8be2e/pkg/kubelet/cm/cpumanager/policy_static.go#L196-L199